src/internal/connector/onedrive/data_collections.go

```diff
@@ -110,3 +123,45 @@ func DataCollections(
 
 	return collections, allExcludes, el.Failure()
 }
+
+// adds data migrations to the collection set.
+func migrationCollections(
```

Comment:
this is (probably?) going to need to work for sharepoint too

hmmm though now I see why this becomes logically complex :/
I'm a little hesitant to have ungated (by backup version or something) folder moves as I'm not sure if we'll ever want to tighten things down re: assumptions about moves (i.e. actually check there was something in the base moved when we thought there should be)

Eventually we may end up passing the previous backup version down to this function since we consume metadata, but we don't have the wiring for it right now to help gate this. Ideally, we should just design some interface that allows passing this (and other) info down to stuff. We could maybe place it on the RestoreCollection, but it feels a tad out of place there since all collections should have the same backup version

It would make me uncomfortable, but we can go ahead with this. We definitely need a ticket to track using this only when we actually need to migrate something though

Reply:
> this is (probably?) going to need to work for sharepoint too

As a general statement?  Or did you have a particular case in mind?

> Eventually we may end up passing the previous backup version down to this function since we consume metadata

I was thinking of doing that for this change already.  Pushing unwarranted migrations sounds like a fantastic footgun.  Wanted to hold off for this PR and re-evaluate today/later, since it's a two-way door.

> passing this (and other) info down to stuff

I waffled on that, myself.  There's an ownership tug-of-war between backup/repo versioning/design and collection sourcing.  If we're going to keep with the collection-based migration, the clearest owner would be the package that produces those collections, and what we'd want to pass down is an indicator of which migration to perform.  Operations is agnostic to the granular details of collection data, and passing in an interface from collections/data_collections is a bit moot.

> We definitely need a ticket to track using this only when we actually need to migrate something though

I'll either follow up today or make a ticket for it by eod.

Reply:
> >this is (probably?) going to need to work for sharepoint too
>
>As a general statement? Or did you have a particular case in mind?

There's probably SharePoint backups out there that will need migrating since I believe we've pointed at least a few folks to that feature so we likely need the ability to handle that

>Wanted to hold off for this PR and re-evaluate today/later, since it's a two-way door.

are you suggesting merge this and fixup later? If we do that I think we should have a solid timeline on when "later" is. As you said, it's a fantastic footgun

Reply:
> There's probably SharePoint backups out there that will need migrating since I believe we've pointed at least a few folks to that feature so we likely need the ability to handle that

I think the plan for sharepoint is not to include migrations, since it's barely been released at all.

> when "later" is.

already patched: https://github.com/alcionai/corso/pull/3137

Reply:
@ashmrtn 3137 has been squashed onto this PR.

Reply:
>I think the plan for sharepoint is not to include migrations, since it's barely been released at all.

It may be better to just plan for SP migrations and handle it if needed. That way there's no surprises down the line

Reply:
> It may be better to just plan for SP migrations and handle it if needed.

Oh, I have it all planned out.  The problem with SP, what we don't have with OD, is that SP would migrate from a value that isn't passed to the caller onto one that is.  The migration itself would be an easy lift on top of this.  It's the additional plumbing specifically to surface data for that migration which I'd like to avoid.